### PR TITLE
Update the resource group tag filter to use server side odata query

### PR DESCRIPTION
**Pull request by hamidtirmizi**
## Description

**Get Resource Group** cmdlet currently performs client side filtering of resource group, and it has a logic bug.

This PR removes client side filtering, and uses the now-supported server side filtering based on odata query.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by hamidtirmizi** [comment]
**Still working on tests**

**Comment by wyunchi-ms** [comment]
Hi @hamidtirmizi how long does the test expect to cost? Do you need any help?

**Comment by hamidtirmizi** [comment]
Tests added. PR ready for review.

**Comment by hamidtirmizi** [comment]
@wyunchi-ms please take a look

File: src/Resources/Resources.Test/Models.ResourceGroups/ResourceClientTests.cs [hunk 1]
```diff
@@ -1153,108 +1152,6 @@ public void GetsAllResourceGroupsWithDetails()
             Assert.Equal(resourceGroup4.Name, actual[3].ResourceGroupName);
         }
 
-        [Fact]
```
Comment (wyunchi-ms) [inline src/Resources/Resources.Test/Models.ResourceGroups/ResourceClientTests.cs]: Hi why these two test cases need to be deleted?

Comment (hamidtirmizi) [reply src/Resources/Resources.Test/Models.ResourceGroups/ResourceClientTests.cs]: These tests were checking that the client side filtering was working on the data retrieved from server.

Now there is no client side filtering. The data is filtered on the server, and we have tests on server side to check appropriate filtering.

**Comment by Geraint87** [comment]
Can we have an example in the docs please? (Or even just here, to get started). I still don't think Get-AzResourceGroup with multiple tags works, or if so, I don't know how.
Example:
`Get-AzResourceGroup -Tag @{AppId = "1"; Environment = "P"};`
Response:
`Invalid tag format. Expect @{tagName = $null} or @{tagName = "tagValue"}`
Thanks!

**Pull request merged**